pkg/cli/image/info/info.go

```diff
@@ -435,12 +404,66 @@ func (o *ImageRetriever) Run() error {
 						Config:        imageConfig,
 						Layers:        layers,
 						Manifest:      srcManifest,
-					}, manifestErr); err != nil {
+					}
+					if err = callbackFn(name, imageRetrieved, manifestErr); err != nil {
 						return err
 					}
 				}
 				return nil
 			})
 		}
 	})
+	if err != nil {
+		return nil, err
+	}
+	return imageRetrieved, nil
+}
+
+// ImageRetrieverOrList returns ImageRetriever or lists os/arch options for manifestList source
```

Comment:
I can update this comment to 
// ImageRetrieverOrList returns ImageRetriever and adds to ManifestListCallback  a msg to distinguish errors from  multiple manifests found from no manifests found, and lists os/arch options for manifestList source if either is the case